### PR TITLE
add multi-orderby continuation token support

### PR DESCRIPTION
**Pull request by xinlian12**
before the change, if the customer is using continuationToken to resume multi-order query, they will get the following exception:"Resuming a multi order by query from a continuation token is not supported yet."

This PR adds the support for it.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/OrderByUtils.java [hunk 1]
```diff
@@ -135,14 +142,27 @@ public PageToItemTransformer(
...
+                                    // If there is a tie in the sort order the documents should be in _rid order in the same direction as the index (given by the backend)
                                     cmp = (continuationTokenRid.getDocument() - ResourceId.tryParse(tOrderByRowResult.getResourceId()).getRight().getDocument());
-
-                                    if (sortOrders.iterator().next().equals(SortOrder.Descending)) {
-                                        cmp = -cmp;
+                                    if ((queryExecutionInfo == null) || queryExecutionInfo.getReverseRidEnabled())
+                                    {
+                                        // If reverse rid is enabled on the backend then fallback to the old way of doing it.
+                                        // So if it's ORDER BY c.age ASC, c.name DESC the _rids are ASC
+                                        // If ti's ORDER BY c.age DESC, c.name DESC the _rids are DESC
```
Comment (mbhaskar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/OrderByUtils.java:151]: typo ti -> it

**Review by mbhaskar — APPROVED** [review]
Thanks also for refactoring the undefined filters.
LGTM

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM

**Comment by xinlian12** [comment]
The failed tests do not relate to the change included in this PR.

Previously passed CI:
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1285291&view=ms.vss-test-web.build-test-results-tab&runId=27457386&resultId=100118&paneView=debug
The only change added in between is a change in comments. 

But still tested locally for following failed tests:
1. loadCachesAndOpenConnectionsToServiceSyncContainer
2. splitQueryContinuationToken
3. queryMetricsForReadAllItemsOfLogicalPartition

Will follow up the following two failed tests separately:
1. queryDocumentsWithTop
2. deleteAllItemsByPartitionKey

**Pull request merged**